src/Servers/Kestrel/Core/src/HttpsConfigurationHelper.cs

```diff
@@ -0,0 +1,227 @@
+    /// </summary>
+    private sealed class ConnectionContextAdapter : ConnectionContext
+    {
+        private readonly BaseConnectionContext _inner;
+
+        public ConnectionContextAdapter(BaseConnectionContext inner) => _inner = inner;
+
+        public override IDuplexPipe Transport
+        {
+            get => throw new NotSupportedException("Not supported by HTTP/3 connections.");
+            set => throw new NotSupportedException("Not supported by HTTP/3 connections.");
+        }
+        public override string ConnectionId
+        {
+            get => _inner.ConnectionId;
+            set => _inner.ConnectionId = value;
+        }
+        public override IFeatureCollection Features => _inner.Features;
+        public override IDictionary<object, object?> Items
+        {
+            get => _inner.Items;
+            set => _inner.Items = value;
+        }
+        public override EndPoint? LocalEndPoint
+        {
+            get => _inner.LocalEndPoint;
+            set => _inner.LocalEndPoint = value;
+        }
+        public override EndPoint? RemoteEndPoint
+        {
+            get => _inner.RemoteEndPoint;
+            set => _inner.RemoteEndPoint = value;
+        }
+        public override CancellationToken ConnectionClosed
+        {
+            get => _inner.ConnectionClosed;
+            set => _inner.ConnectionClosed = value;
+        }
+        public override ValueTask DisposeAsync() => _inner.DisposeAsync();
+    }
+
+    // TODO (acasey): Make this public and move it to Connection.Abstractions for use in UseQuic?
+    internal interface IEnabler
+    {
+        void Enable(IHttpsConfigurationHelper httpsConfigurationHelper);
+    }
+
+    internal sealed class Enabler : IEnabler
+    {
+        private readonly IHostEnvironment _hostEnvironment;
+        private readonly ILogger<KestrelServer> _serverLogger;
+        private readonly ILogger<HttpsConnectionMiddleware> _httpsLogger;
+
+        public Enabler(
+            IHostEnvironment hostEnvironment,
+            ILogger<KestrelServer> serverLogger,
+            ILogger<HttpsConnectionMiddleware> httpsLogger)
+        {
+            Logger.WriteLine("Constructing Enabler");
+            _hostEnvironment = hostEnvironment;
+            _serverLogger = serverLogger;
+            _httpsLogger = httpsLogger;
+        }
+
+        public void Enable(IHttpsConfigurationHelper httpsConfigurationHelper)
+        {
+            httpsConfigurationHelper.Enable(_hostEnvironment, _serverLogger, _httpsLogger);
+        }
+    }
+
+    internal static class Logger
+    {
+        private static readonly StringBuilder _builder = new StringBuilder();
+
+        public static void WriteLine(string msg)
+        {
+            _builder.AppendLine(msg);
+        }
+
+        public static string GetLog()
+        {
+            return _builder.ToString();
+        }
+    }
```

Comment:
Why not write to ILogger? Lots of tests pass the ITestOutputHelper to the logger and then write all the test logs to output.

Reply:
I like the idea of having good logs captured from HTTP/TLS types. IMO Kestrel doesn't log enough during startup today.

Reply:
This was just a hack to debug some test failures, but I'm totally on board with adding more real logging.

Reply:
Sorry for not leaving a comment on the type.